datafusion/physical-expr/src/math_expressions.rs

```diff
@@ -287,6 +288,16 @@ pub fn random(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     Ok(ColumnarValue::Array(Arc::new(array)))
 }
 
+macro_rules! round_downcast_decimal_places {
+    ($dp:expr) => {
+        $dp.try_into().map_err(|_| {
+            DataFusionError::Execution(
+                "round function requires decimal_places as 32 bit integers".to_string(),
+            )
+        })
+    };
+}
+
```

Comment:
The input/output types are always i64/i32. This isn't necessary to be a macro, I guess.

Reply:
You can have a inline function instead.